block_evasys_sync.php

```diff
@@ -48,47 +49,112 @@ public function get_content() {
-                            format_string($survey->amountOfCompletedForms));
-                }
-                $this->content->text .= html_writer::alist($outputsurveys, null, 'ol');
 
+            if (!$this->getmode($this->page->course->category)) {
+                $href = new moodle_url('/blocks/evasys_sync/sync.php');
+            } else {
+                $href = new moodle_url('/course/view.php',
+                                       array('id' => $this->page->course->id, "evasyssynccheck" => true));
             }
-            $this->content->text .= html_writer::div(html_writer::span(
-                get_string('countparticipants', 'block_evasys_sync'), 'emphasize') . ' ' .
-                format_string($evasyssynchronizer->get_amount_participants()));
-            $href = new moodle_url('/blocks/evasys_sync/sync.php', array('courseid' => $this->page->course->id));
-            $this->content->text .= $OUTPUT->single_button($href, get_string('invitestudents', 'block_evasys_sync'), 'post' );
+            // Begin form.
+            $this->content->text .= "<form action='$href' method='post' id='evasys_block_form'>";
+            $this->content->text .= "<input type='hidden' name='sesskey' value='" . sesskey() . "'>";
+            $this->content->text .= "<input type='hidden' name='courseid' value='" . $this->page->course->id . "'>";
 
+            $i = 0;
+            // Output for each Evasys-course mapped to this moodle course.
+            foreach ($evasyscourseids as $evasyscourseid) {
+                $this->content->text .= html_writer::div(html_writer::span(
+                                                             get_string('evacourseid', 'block_evasys_sync'), 'emphasize') . ' ' . $evasyscourseid);
+                $this->content->text .= html_writer::div(html_writer::span(
+                                                             get_string('countparticipants', 'block_evasys_sync'), 'emphasize') . ' ' .
+                                                         format_string($evasyssynchronizer->get_amount_participants($evasyscourseid)));
+                $this->content->text .= html_writer::div(get_string('surveys', 'block_evasys_sync'), 'emphasize');
+                $outputsurveys = array();
+                $surveys = $evasyssynchronizer->get_surveys($evasyscourseid);
+
+                if (empty($surveys)) {
+                    $this->content->text .= get_string('nosurveys', 'block_evasys_sync');
+                } else {
+                    $readonly = "readonly";
+                    foreach ($surveys as &$survey) {
+                        // Output for each of those surveys :D.
+                        $prefills = \block_evasys_sync\evaluationperiod_survey_allocation::get_record(array('survey' => $survey->id));
+                        // Find if dates have been defined before and should be prefilled.
+                        if (!$prefills) {
```

Comment:
Here you can better work with the timestamps and time(), so you don't have to compare strings of timestamps, and convert them into a string when you want to show them.